test/contracts-network/colony-funding.js

```diff
@@ -549,5 +562,264 @@ contract("Colony Funding", (accounts) => {
+      const receiverAddress = await colonyNetwork.getDomainTokenReceiverAddress(colony.address, 2);
+
+      // Send 100 wei
+      await web3.eth.sendTransaction({ from: MANAGER, to: receiverAddress, value: 100, gas: 1000000 });
+
+      const domain = await colony.getDomain(2);
+      const domainPotBalanceBefore = await colony.getFundingPotBalance(domain.fundingPotId, ethers.constants.AddressZero);
+      const nonRewardPotsTotalBefore = await colony.getNonRewardPotsTotal(ethers.constants.AddressZero);
+
+      // Claim the funds
+
+      const tx = await colony.claimDomainFunds(ethers.constants.AddressZero, 2);
+      await expectEvent(tx, "DomainFundsClaimed", [MANAGER, ethers.constants.AddressZero, 2, 1, 99]);
+
+      const domainPotBalanceAfter = await colony.getFundingPotBalance(domain.fundingPotId, ethers.constants.AddressZero);
+      const nonRewardPotsTotalAfter = await colony.getNonRewardPotsTotal(ethers.constants.AddressZero);
+
+      // Check the balance of the domain
+      expect(domainPotBalanceAfter.sub(domainPotBalanceBefore)).to.eq.BN(99);
+      expect(nonRewardPotsTotalAfter.sub(nonRewardPotsTotalBefore)).to.eq.BN(99);
+    });
+
+    it("should allow a token to be directly sent to a domain", async () => {
+      // Get address for domain 2
+      await colony.addDomain(1, UINT256_MAX, 1);
+      const receiverAddress = await colonyNetwork.getDomainTokenReceiverAddress(colony.address, 2);
+
+      // Send 100 wei
+      await otherToken.mint(receiverAddress, 100);
+
+      const domain = await colony.getDomain(2);
+      const domainPotBalanceBefore = await colony.getFundingPotBalance(domain.fundingPotId, otherToken.address);
+      const nonRewardPotsTotalBefore = await colony.getNonRewardPotsTotal(otherToken.address);
+
+      // Claim the funds
+      const tx = await colony.claimDomainFunds(otherToken.address, 2);
+      await expectEvent(tx, "DomainFundsClaimed", [MANAGER, otherToken.address, 2, 1, 99]);
+
+      const domainPotBalanceAfter = await colony.getFundingPotBalance(domain.fundingPotId, otherToken.address);
+      const nonRewardPotsTotalAfter = await colony.getNonRewardPotsTotal(otherToken.address);
+
+      // Check the balance of the domain
+      expect(domainPotBalanceAfter.sub(domainPotBalanceBefore)).to.eq.BN(99);
+      expect(nonRewardPotsTotalAfter.sub(nonRewardPotsTotalBefore)).to.eq.BN(99);
+    });
+
+    it("when receiving native (reputation-earning) token, if no approval present for domain, all are received by root domain", async () => {
+      // Get address for domain 2
+      await colony.addDomain(1, UINT256_MAX, 1);
+      const receiverAddress = await colonyNetwork.getDomainTokenReceiverAddress(colony.address, 2);
+      await colony.mintTokens(WAD.muln(100));
+      await colony.claimColonyFunds(token.address);
+      const domain1 = await colony.getDomain(1);
+
+      // Pay the tokens to the domain
+      await setupClaimedExpenditure({
```

Comment:
Is this necessary? Cant you just give the user some native tokens directly?

Reply:
Just let me know how and I will :smile:

Reply:
Would `colony.mintTokensFor` not work? IIRC we've discussed this in the past and came up with an alternative to using expenditures, but maybe I'm hallucinating.

Reply:
Hah, fair play! I had no recollection of that function existing. It seems as if it was introduced for CoinMachine, which a) we're no longer really using, and b) no longer actually uses that function. I would like to propose deprecating and removing it, rather than starting to use it 😅

Reply:
I can live with that.

How about using `giveUserCLNYTokens` instead? We've definitely been here before...

Reply:
Happy with your solution! Thanks for entertaining me.